core/tests/dispatch_integration_tests.rs

```diff
@@ -1077,47 +1024,35 @@ fn network_status_port_established_lost_dropped_connection() {
-    let (status_counter, _scf) = local_system.create_and_register(NetworkStatusCounter::new);
-    status_counter.on_definition(|c| {
-        local_system.connect_network_status_port(&mut c.network_status_port);
-    });
-    local_system.start(&status_counter);
+    let status_counter = create_and_run_status_counter(&local_system);
 
     // Create a pinger ponger pair such that the Network will be used.
-    let (ponger, _pof) = local_system.create_and_register(PongerAct::new_lazy);
-    let ponger_path = local_system
-        .register_by_alias(&ponger, "ponger")
-        .wait_expect(Duration::from_millis(1000), "Ponger failed to register!");
-    local_system.start(&ponger);
-    let (pinger, _pif) =
-        remote_system.create_and_register(move || PingerAct::new_lazy(ponger_path));
-    let pinger_path = remote_system
-        .register_by_alias(&ponger, "ponger")
-        .wait_expect(Duration::from_millis(1000), "Ponger failed to register!");
-    remote_system.start(&pinger);
-    // The systems establish a connection and the pings/pongs are sent
-    thread::sleep(Duration::from_millis(5000));
+    let (_, ponger_path) = create_and_run_lazy_ponger(&remote_system);
+    let (_, pinger_done_future) = create_and_run_lazy_pinger(&local_system, &ponger_path);
 
-    // Inspect live sockets
-    // thread::sleep(Duration::from_millis(2 * 60 * 1000)); // "2MSL" to drop the socket
+    pinger_done_future
+        .wait_timeout(Duration::from_secs(10))
+        .expect("Pinger should complete");
 
     // Shutdown the remote system and wait for the connection to be lost and dropped by local_system
     let _ = remote_system.kill_system();
-    thread::sleep(Duration::from_millis(2500));
+    thread::sleep(Duration::from_millis(2000));
 
-    let pinger_path_clone = pinger_path.clone();
     // Make sure the local_system discovers the lost connection
-    let (failing_pinger, _pif) =
-        local_system.create_and_register(move || PingerAct::new_lazy(pinger_path_clone));
-    local_system.start(&failing_pinger);
+    let (pinger2, reg_future) =
+        local_system.create_and_register(|| PingerAct::new_lazy(ponger_path.clone()));
+    reg_future.wait_expect(Duration::from_millis(1000), "Pinger failed to register!");
+    local_system.start(&pinger2);
 
+    thread::sleep(Duration::from_millis(2000));
     // Make sure the local_system discovers the lost connection on Linux....
-    let (failing_pinger2, _pif) =
-        local_system.create_and_register(move || PingerAct::new_lazy(pinger_path));
-    local_system.start(&failing_pinger2);
+    let (pinger3, reg_future) =
+        local_system.create_and_register(|| PingerAct::new_lazy(ponger_path.clone()));
+    reg_future.wait_expect(Duration::from_millis(1000), "Pinger failed to register!");
+    local_system.start(&pinger3);
 
     // Wait for connection to be dropped
-    thread::sleep(Duration::from_millis(12000));
+    thread::sleep(Duration::from_millis(10000));
```

Comment:
I mean...10s is better than 12s. But you know what's even better? Yes, you guessed right...no unconditional waiting! :P